HabitRPG/TableViewDataSources/FixedSizeDataSource.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  FixedSizeDataSource.swift
+//  Habitica
+//
+//  Created by Elliot Schrock on 10/17/17.
+//  Copyright © 2017 HabitRPG Inc. All rights reserved.
+//
+
+import UIKit
+
+protocol FixedSizeDataSourceItem {
+    func cellIdentifier() -> String
+    func cellClass() -> UITableViewCell.Type
+    func configureCell(_ cell: UITableViewCell)
+}
+
+// MARK: -
+
+class FixedSizeDataSourceSection {
+    var title: String?
+    var items: [FixedSizeDataSourceItem]?
+}
+
+// MARK: -
+
+class FixedSizeDataSource: NSObject, UITableViewDataSource {
```

Comment:
What does "FixedSize" refer to here? Since it looks like it can still be updated, so the content wouldn't be fixed.

Reply:
Yeah, naming things is hard :/ 

This pattern (using an array of helper models to dictate the cell types and configuration) seems to work best when two things are true: you're going to be displaying many different types of cells, and you're not going to implement an infinite scroll. These are tied together: if there are as many cell types as there are cells, *and* you have many cell types, the table will have to store all those cells in memory and 'dequeue'ing won't be saving you much of anything memory-wise. Therefore, it's important to limit the size of a data source of this nature, which is kind of what I was trying to express with the name. That is, `FixedSize` refers to the need to not let the number of cells grow and grow and grow... even though technically you can.

BUT, looking at it now, I'm realizing that a better name might be something along the lines of `MixedCellTypeDataSource`, since it's the part about having multiple types of cells displayed in the same table that drives much of the advantages and disadvantages of this approach. Would you like me to change it to that, or something similar?